frontend/src/mixins/shootItem.js

```diff
@@ -236,6 +236,85 @@ export const shootItem = {
     },
     maintenancePreconditionSatisfiedMessage () {
       return get(this.maintenancePreconditionSatisfiedConstraint, 'message', 'It may not be safe to trigger maintenance for this cluster')
+    },
+    caCertificateValiditiesAcceptableConstraint () {
+      const constraints = this.shootConstraints
+      return find(constraints, ['type', 'CACertificateValiditiesAcceptable'])
+    },
+    isCACertificateValiditiesAcceptable () {
+      const status = get(this.caCertificateValiditiesAcceptableConstraint, 'status', 'True')
+      return status !== 'False'
+    },
+    caCertificateValiditiesAcceptableMessage () {
+      return get(this.caCertificateValiditiesAcceptableConstraint, 'message', 'There is at least one CA certificate which expires in less than 1y. Consider schduling a Certificate Authorities Rotation for this cluster')
+    },
+    shootStatus () {
+      return get(this.shootItem, 'status', {})
+    },
+    shootStatusCredentialsRotation () {
+      return get(this.shootStatus, 'credentials.rotation', {})
+    },
+    shootStatusCredentialsRotationAggregatedPhase () {
+      let preparedRotationsCount = 0
+      let completedPhasesCount = 0
+      const unpreparedRotations = []
+      const keys = Object.keys(this.shootStatusCredentialsRotation)
+      for (let i = 0; i < keys.length; i++) {
+        // use simple for loop to support early exit (immediately return in case of progressing phase)
+        const rotationKey = keys[i]
+        const rotationStatus = this.shootStatusCredentialsRotation[rotationKey]
```

Comment:
```suggestion
      for (const [rotationKey, rotationStatus] of Object.entries(this.shootStatusCredentialsRotation)) {
        // use simple for loop to support early exit (immediately return in case of progressing phase)
```